widgets/src/components/common/WidgetConfig.js

```diff
@@ -0,0 +1,104 @@
+import React, { useEffect, useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import Button from '@material-ui/core/Button';
+import { Close, Check } from '@material-ui/icons';
+import CircularProgress from '@material-ui/core/CircularProgress';
+
+import { getPageWidget, putPageWidget } from 'api/app-builder/pages';
+
+const WidgetConfig = ({ children, config: passedConfig, ...rest }) => {
+  const [loaded, setLoaded] = useState(false);
+  const [code, setCode] = useState({});
+  const [config, setConfig] = useState({});
+  const [saving, setSaving] = useState(false);
+  const [saved, setSaved] = useState({});
```

Comment:
can't it be a boolean instead?

Reply:
It shows different icon in case there is an error. It's just simplest way to give feedback to developer without connecting notifications or adding alerts - does not affect development that much